controllers/pipelines/runconfiguration_controller.go

```diff
@@ -98,7 +99,8 @@ func (r *RunConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	newStatus.SynchronizationState = newSynchronizationState
 	newStatus.Provider = desiredProvider
 
-	if newStatus != runConfiguration.Status {
+
+	if !reflect.DeepEqual(newStatus, runConfiguration.Status) {
```

Comment:
RunConfigurationStatus is no longer comparable by using `==`. 
Reference: https://pkg.go.dev/reflect#DeepEqual